src/utils/beacon/BeaconPeers.tsx

```diff
@@ -52,64 +49,83 @@ export const BeaconPeers = () => {
     );
   }
 
-  return <PeersDisplay peerInfos={peersWithId} removePeer={removePeer} />;
+  return (
+    <Box>
+      {peersWithId.map(peerInfo => (
+        <PeersDisplay key={peerInfo.senderId} peerInfo={peerInfo} />
```

Comment:
senderId is a non-unique property. if we have two connections to the same dApp on the same network from 2 different addressed then we have a duplicate key here. It's better to use just `id`

Reply:
In this case we will have duplicating connections (incorrectly assigned to dApps) because we use senderId as an identifier for connection in BeaconSlice

Should we migrate slice to senderId to avoid that?

Reply:
not sure I follow 🤔 
[here](https://github.com/trilitech/umami-v2/pull/931/files#diff-1cc2311e2ccff0d9f9edf41bfd474d21ab66619e4677e2fe888035acb0c66e10R7) it's said that the beacon slice is a nested map essentially. so, if we have a dApp with id `lolkek` it'd be
```javascript
{
  lolkek: {
         address1: "mainnet",
         address2: "mainnet"
     }
}
```

right?

Reply:
My understanding is that senderId is dApp's id and it's unique for each dApp. If that's correct, the existing solution should work as far as I understand.

We use request.senderId to identify dApps in a beaconSlice:
```addConnectionToBeaconSlice(request.senderId, account.address.pkh, request.network.type)``` in [PermissionRequestModal](https://github.com/trilitech/umami-v2/blob/main/src/utils/beacon/PermissionRequestModal.tsx#L64)

In [line 55](https://github.com/trilitech/umami-v2/blob/b58cf4a8bfe8846aa72afb4d9cb309356d3b5d21/src/utils/beacon/BeaconPeers.tsx#L55) we create one block per dApp to list dApp's connections (block identified by a `senderId`).
```
    <Box>
      {peersWithId.map(peerInfo => (
        <PeersDisplay key={peerInfo.senderId} peerInfo={peerInfo} />
      ))}
    </Box>
```

In [PeersDisplay](https://github.com/trilitech/umami-v2/blob/b58cf4a8bfe8846aa72afb4d9cb309356d3b5d21/src/utils/beacon/BeaconPeers.tsx#L68-L91) for given dApp we list connection tiles for each dApp's connection (one if no connections is saved, or one per connected account).
Each list is treated separately, so having `peerInfo.name + accountPkh` as an id should be safe: it's unique across connected accounts for given dApp.

```
  if (connectedAccounts.length === 0) {
    return (
      <Fragment key={peerInfo.name}>
        <Divider />
        <PeerRow peerInfo={peerInfo} />
      </Fragment>
    );
  }

  return (
    <Box>
      {connectedAccounts.map(accountPkh => (
        <Fragment key={peerInfo.name + accountPkh}>
          <Divider />
          <PeerRow accountPkh={accountPkh} peerInfo={peerInfo} />
        </Fragment>
      ))}
    </Box>
  );
```

Reply:
hmm, I think I see why it works with `senderId`.  senderId is unique not for a dApp, but rather a beacon instance of a dapp.
I connected my wallet to objkt using 2 different browsers and that's what I've got in the peers list:
<img width="553" alt="Screenshot 2024-03-11 at 14 38 10" src="https://github.com/trilitech/umami-v2/assets/129749432/afe2fb74-c5b5-443c-94e6-0b59d1b8c1c8">
so, yeah, senderID is fine :+1: